client/internal/clientcommon.go

```diff
@@ -67,30 +65,67 @@ func NewClientCommon(logger types.Logger, sender Sender) ClientCommon {
 	return ClientCommon{Logger: logger, sender: sender, stoppedSignal: make(chan struct{}, 1)}
 }
 
+func (c *ClientCommon) hasCapability(capability protobufs.AgentCapabilities) bool {
+	return c.ClientSyncedState.Capabilities()&capability != 0
+}
+
+func (c *ClientCommon) validateCapabilities(capabilities protobufs.AgentCapabilities) error {
+	if capabilities&protobufs.AgentCapabilities_AgentCapabilities_ReportsHealth != 0 && c.ClientSyncedState.Health() == nil {
+		return ErrHealthMissing
+	}
+	if capabilities&protobufs.AgentCapabilities_AgentCapabilities_ReportsAvailableComponents != 0 && c.ClientSyncedState.AvailableComponents() == nil {
+		return ErrAvailableComponentsMissing
+	}
+	if c.PackagesStateProvider != nil {
+		if (capabilities&protobufs.AgentCapabilities_AgentCapabilities_AcceptsPackages == 0) ||
+			(capabilities&protobufs.AgentCapabilities_AgentCapabilities_ReportsPackageStatuses == 0) {
+			return ErrAcceptsPackagesNotSet
+		}
+	} else {
+		if capabilities&protobufs.AgentCapabilities_AgentCapabilities_AcceptsPackages != 0 ||
+			capabilities&protobufs.AgentCapabilities_AgentCapabilities_ReportsPackageStatuses != 0 {
+			return ErrPackagesStateProviderNotSet
+		}
+	}
+	return nil
+}
+
 // PrepareStart prepares the client state for the next Start() call.
 // It returns an error if the client is already started, or if the settings are invalid.
 func (c *ClientCommon) PrepareStart(
-	_ context.Context, settings types.StartSettings,
+	ctx context.Context, settings types.StartSettings,
 ) error {
 	if c.isStarted {
 		return errAlreadyStarted
 	}
-
-	c.Capabilities = settings.Capabilities
-
-	// According to OpAMP spec this capability MUST be set, since all Agents MUST report status.
-	c.Capabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsStatus
+	// Deprecated: Use client.SetCapabilities() instead.
+	if settings.Capabilities != 0 {
+		c.Logger.Errorf(ctx, "settings.Capabilities is deprecated, use client.SetCapabilities() instead")
+		capabilities := settings.Capabilities
+		// According to OpAMP spec this capability MUST be set, since all Agents MUST report status.
+		capabilities |= protobufs.AgentCapabilities_AgentCapabilities_ReportsStatus
+		if err := c.ClientSyncedState.SetCapabilities(&capabilities); err != nil {
+			return err
+		}
+	}
+	if c.ClientSyncedState.Capabilities() == 0 {
+		c.Logger.Errorf(ctx, "you must call client.SetCapabilities before start.")
+		capabilities := protobufs.AgentCapabilities_AgentCapabilities_ReportsStatus
+		if err := c.ClientSyncedState.SetCapabilities(&capabilities); err != nil {
+			return err
+		}
+		// Eventually we will error here.
```

Comment:
What does "Eventually" mean? Do we need this to be an error?

Reply:
this was in reference to your comment [here](https://github.com/open-telemetry/opamp-go/pull/366#discussion_r2120874171). I can open an issue and just reference that instead.